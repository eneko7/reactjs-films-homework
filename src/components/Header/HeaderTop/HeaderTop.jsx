import React from 'react';
import PropTypes from 'prop-types';
import style from './HeaderTop.scss';

class HeaderTop extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.searchFilms = this.searchFilms.bind(this);
  }

  onChange(e) {
    const { searchFilm, fetchFilmsPopular } = this.props;
    if (e.target.value !== '') {
      searchFilm(e.target.value);
    } else {
      fetchFilmsPopular();
    }
  }

  searchFilms() {
    const { searchFilm } = this.props;
    searchFilm(document.getElementById('search_input').value);
  }

  render() {
    return (
      <div className={style.header_top}>
        <span className={style.header_top_span}>FILMS</span>
        <div className={style.header_top_search}>
          <input type="text" id="search_input" className={style.header_top_search_input} onChange={this.onChange} placeholder="Searh..." />
          <button className={style.header_top_search_label} onClick={this.searchFilms} type="button" />
        </div>
      </div>
    );
  }
}
HeaderTop.propTypes = {
  fetchFilmsPopular: PropTypes.func.isRequired,
  searchFilm: PropTypes.func.isRequired,
};
export default HeaderTop;
