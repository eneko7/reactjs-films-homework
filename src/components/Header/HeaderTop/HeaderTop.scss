.header_top {
  .header_top_span {
    display: block;
    float: left;
    font-size: 35px;
    letter-spacing: 1px;
    line-height: 45px;
    color: #fff;
    font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
    font-weight: 800;
    position: relative;
    z-index: 3;
    margin-left: 7%;
  }

  .header_top_search {
    display: block;
    float: right;
    width: 390px;
    margin-right: 80px;
    background: rgba(93, 155, 211, 0.01);
    position: relative;
    z-index: 3;

    .header_top_search_label {
      color: #fff;
      background-image: url(../../../images/magnifying-glass.png);
      width: 20px;
      height: 20px;
      background-position: center;
      background-size: contain;
      display: inline-block;
      position: absolute;
      right: 4%;
      top: 25%;
      cursor: pointer;
    }

    .header_top_search_input {
      width: 100%;
      height: 45px;
      border: 3.5px solid #fff;
      border-radius: 8%;
      position: relative;
      background: none;
      font-size: 18px;
      letter-spacing: 1px;
      line-height: 30px;
      color: #fff;
      font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
      padding-left: 15px;
      padding-right: 15px;
      text-shadow: 0.5px 0.5px #000;
      display: inline-block;
    }

    .header_top_search_input::placeholder {
      color: #fff;
      opacity: 1;
    }

    .header_top_search_input:focus {
      outline: none;
    }
  }
}
