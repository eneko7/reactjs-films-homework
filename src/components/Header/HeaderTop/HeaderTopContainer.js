import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import HeaderTop from './HeaderTop';
import { saveSearchingWord, searchFilm } from '../../../modules/seacrhFilms/searchFilmsActions';
import { fetchFilmsPopular } from '../../../modules/films/filmsActions';
import getSearchingWord from '../../../modules/seacrhFilms/searchFilmsSelectors';

const mapStateToProps = state => ({
  searchedWord: getSearchingWord(state),
});

const mapDispatchToProps = {
  saveSearchingWord,
  searchFilm,
  fetchFilmsPopular,
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(HeaderTop));
