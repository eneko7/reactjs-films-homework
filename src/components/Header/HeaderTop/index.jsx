import HeaderTop from './HeaderTopContainer';

export default HeaderTop;
