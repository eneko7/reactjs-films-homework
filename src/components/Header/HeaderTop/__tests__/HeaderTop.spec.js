import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HeaderTop from '../index';

describe('HeaderTop', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HeaderTop />);
    it('renders correctly', () => {
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});