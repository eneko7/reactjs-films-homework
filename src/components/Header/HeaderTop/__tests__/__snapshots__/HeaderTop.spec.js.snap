// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTop renders correctly 1`] = `
<div
  className="header_top"
>
  <span
    className="header_top_span"
  >
    FILMS
  </span>
  <div
    className="header_top_search"
  >
    <input
      className="header_top_search_input"
      id="search_input"
      placeholder="Searh..."
      type="text"
    />
    <label
      className="header_top_search_label"
      htmlFor="search_input"
    />
  </div>
</div>
`;
